internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp

```diff
@@ -472,9 +472,23 @@ ChunkedSegmentSealedImpl::LoadFieldData(FieldId field_id, FieldDataInfo& data) {
             update_row_count(num_rows);
         }
 
-        if (!generate_interim_index(field_id)) {
+        bool use_interim_index = generate_interim_index(field_id);
+        if (!use_interim_index) {
             std::unique_lock lck(mutex_);
             set_bit(field_data_ready_bitset_, field_id, true);
+        } else {
+            std::unique_lock lck(mutex_);
+            auto field_indexing =
+                vector_indexings_.get_field_indexing(field_id);
+            auto interim_index = dynamic_cast<index::VectorIndex*>(
+                field_indexing->indexing_.get());
+            if (interim_index->HasRawData()) {
+                fields_.erase(field_id);
```

Comment:
Is the `fields` used in other places? In another words, does it safe to erase field from it